x/trigger/simulation/genesis.go

```diff
@@ -0,0 +1,140 @@
+	amount := int64(simtypes.RandIntBetween(r, 100, 1000))
+	return &banktypes.MsgSend{
+		FromAddress: from,
+		ToAddress:   to,
+		Amount:      sdk.NewCoins(sdk.NewInt64Coin(pioconfig.GetProvenanceConfig().BondDenom, amount)),
+	}
+}
+
+// GetRandomTrigger returns a random trigger
+func GetRandomTrigger(r *rand.Rand, simState *module.SimulationState) types.Trigger {
+	account := int64(simtypes.RandIntBetween(r, 1, 2))
+	event := GetRandomEvent(r, simState.GenTimestamp.UTC())
+	action := GetRandomAction(r, simState.Accounts[0].Address.String(), simState.Accounts[account].Address.String())
+
+	actions, err := sdktx.SetMsgs([]sdk.Msg{action})
+	if err != nil {
+		panic("SetMsgs failed for GetRandomTrigger")
+	}
+	anyEvent, err := codectypes.NewAnyWithValue(event)
+	if err != nil {
+		panic("NewAnyWithValue failed for GetRandomTrigger")
+	}
+	return types.NewTrigger(1, simState.Accounts[0].Address.String(), anyEvent, actions)
+}
+
+// GetRandomQueuedTrigger returns a random queued trigger
+func GetRandomQueuedTrigger(_ *rand.Rand, simState *module.SimulationState) types.QueuedTrigger {
+	var event types.TriggerEventI = &types.BlockHeightEvent{BlockHeight: 100000}
+	var action sdk.Msg = &banktypes.MsgSend{
+		FromAddress: simState.Accounts[0].Address.String(),
+		ToAddress:   simState.Accounts[1].Address.String(),
+		Amount:      sdk.NewCoins(sdk.NewInt64Coin(pioconfig.GetProvenanceConfig().BondDenom, 100)),
+	}
+
+	actions, err := sdktx.SetMsgs([]sdk.Msg{action})
+	if err != nil {
+		panic("SetMsgs failed for GetRandomQueuedTrigger")
+	}
+	anyEvent, err := codectypes.NewAnyWithValue(event)
+	if err != nil {
+		panic("NewAnyWithValue failed for GetRandomQueuedTrigger")
+	}
+	trigger := types.NewTrigger(2, simState.Accounts[0].Address.String(), anyEvent, actions)
+
+	now := simState.GenTimestamp
+
+	return types.NewQueuedTrigger(trigger, now.UTC(), 1)
+}
+
+// RandomizedGenState generates a random GenesisState for trigger
+func RandomizedGenState(simState *module.SimulationState) {
+	var triggerID uint64
+	simState.AppParams.GetOrGenerate(
+		simState.Cdc, TriggerID, &triggerID, simState.Rand,
+		func(r *rand.Rand) { triggerID = TriggerIDFn(r) },
+	)
+
+	var queueStart uint64
+	simState.AppParams.GetOrGenerate(
+		simState.Cdc, QueueStart, &queueStart, simState.Rand,
+		func(r *rand.Rand) { queueStart = QueueStartFn(r) },
+	)
+
+	triggers := []types.Trigger{
+		GetRandomTrigger(simState.Rand, simState),
+	}
+	queuedTriggers := []types.QueuedTrigger{
+		GetRandomQueuedTrigger(simState.Rand, simState),
+	}
+
+	gasLimits := []types.GasLimit{}
+	for _, trigger := range triggers {
+		gasLimits = append(gasLimits, types.GasLimit{
+			TriggerId: trigger.GetId(),
+			Amount:    1000000,
+		})
+	}
+	for _, item := range queuedTriggers {
+		gasLimits = append(gasLimits, types.GasLimit{
+			TriggerId: item.Trigger.Id,
+			Amount:    1000000,
+		})
+	}
+
+	genesis := types.NewGenesisState(triggerID, queueStart, triggers, gasLimits, queuedTriggers)
```

Comment:
For the sims, you want `RandomizedGenState` to create a valid genesis state.

It's possible in here, though, to have duplicated trigger ids or have triggers with ids greater than `triggerID`.